pkg/bootstrap/render.go

```diff
@@ -19,214 +19,183 @@ import (
+	}
 }
 
-type KlusterletManifestsConfig struct {
-	InstallMode operatorv1.InstallMode
-
-	ClusterName                  string
-	KlusterletNamespace          string
-	KlusterletClusterAnnotations map[string]string
-	BootstrapKubeconfig          []byte
-
-	// PriorityClassName is the name of the PriorityClass used by the klusterlet and operator
-	PriorityClassName string
-
-	// Used to determine whether mc is a localcluster.
-	ManagedCluster *clusterv1.ManagedCluster
-
-	klusterletconfig *klusterletconfigv1alpha1.KlusterletConfig
+func newKlusterletChartConfig(installMode operatorv1.InstallMode,
+	clusterName string, bootstrapKubeConfig []byte) *chart.KlusterletChartConfig {
+	allowPrivilegeEscalation := false
+	privileged := false
+	runAsNonRoot := true
+	readOnlyRootFilesystem := true
+
+	chartConfig := &chart.KlusterletChartConfig{
+		ReplicaCount: 1,
+		Images: chart.ImagesConfig{
+			ImagePullPolicy: corev1.PullIfNotPresent,
+			ImageCredentials: chart.ImageCredentials{
+				// set true by default
+				CreateImageCredentials: true,
+			},
+		},
+		// the release ns and the agent ns are same, so disable to generate ns.
+		CreateNamespace: true,
+		PodSecurityContext: corev1.PodSecurityContext{
+			RunAsNonRoot: &runAsNonRoot,
+		},
+		SecurityContext: corev1.SecurityContext{
+			Capabilities: &corev1.Capabilities{
+				Drop: []corev1.Capability{"ALL"},
+			},
+			Privileged: &privileged,
 
-	generateImagePullSecret bool // by default is true, in hosted mode, it will be set false
-}
+			RunAsNonRoot:             &runAsNonRoot,
+			ReadOnlyRootFilesystem:   &readOnlyRootFilesystem,
+			AllowPrivilegeEscalation: &allowPrivilegeEscalation,
+		},
+		Resources: corev1.ResourceRequirements{
+			Limits: corev1.ResourceList{
+				corev1.ResourceMemory: resource.MustParse("2Gi"),
+			},
+			Requests: corev1.ResourceList{
+				corev1.ResourceMemory: resource.MustParse("64Mi"),
+				corev1.ResourceCPU:    resource.MustParse("50m"),
+			},
+		},
+		NodeSelector: map[string]string{},
+		Tolerations:  []corev1.Toleration{},
+		Affinity:     corev1.Affinity{},
+		Klusterlet: chart.KlusterletConfig{
+			ClusterName: clusterName,
+			Mode:        installMode,
+		},
+		PriorityClassName:         "",
+		EnableSyncLabels:          false,
+		BootstrapHubKubeConfig:    string(bootstrapKubeConfig),
+		ExternalManagedKubeConfig: "",
+		NoOperator:                false,
+	}
 
-func NewKlusterletManifestsConfig(installMode operatorv1.InstallMode,
-	clusterName string, bootstrapKubeconfig []byte) *KlusterletManifestsConfig {
-	return &KlusterletManifestsConfig{
-		InstallMode:             installMode,
-		ClusterName:             clusterName,
-		BootstrapKubeconfig:     bootstrapKubeconfig,
-		generateImagePullSecret: true,
+	if chartConfig.Klusterlet.Mode == operatorv1.InstallModeHosted ||
+		chartConfig.Klusterlet.Mode == operatorv1.InstallModeSingletonHosted {
+		chartConfig.CreateNamespace = false
```

Comment:
Should the noOperator be set to true in hosted mode?

Reply:
yes,  noOperator is set  here https://github.com/stolostron/managedcluster-import-controller/pull/603/files#diff-34fd40905488f5ef71014d75f1d60e6a44679858fc341862946485ae03b11fddR183 there is a general function named installNoOperator  to set noOperator